### PR TITLE
feat: Create RegisterValidationResponse for authController

### PR DESCRIPTION
**Pull request by drizlye0**
## Description
Create better error response for authController validator

Error response:
image
## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update

### CODE REVIEW

**Comment by Flussen** [comment]
Recent changes in the develop branch generated conflicts

**Comment by matdevcoder** [comment]
@drizlye0 @Flussen I see it well for the moment but maybe in the future we can provide a way to insert it or embeded this `RegisterValidationErr()` in another error to make it more standable. @Flussen what are you thought?

Regarding to the code, maybe you could add a validation to add a type assertion check before getting the errors from the for range in `RegisterValidationErr()`, something like this:

```go
validationErrors, ok := errs.(validator.ValidationErrors)
if !ok {
     return c.Status(fiber.StatusBadRequest).JSON(err)
 }
```

**Comment by Flussen** [comment]
> @drizlye0 @Flussen I see it well for the moment but maybe in the future we can provide a way to insert it or embeded this `RegisterValidationErr()` in another error to make it more standable. @Flussen what are you thought?
> 
> Regarding to the code, maybe you could add a validation to add a type assertion check before getting the errors from the for range in `RegisterValidationErr()`, something like this:
> 
> ```go
> validationErrors, ok := errs.(validator.ValidationErrors)
> if !ok {
>      return c.Status(fiber.StatusBadRequest).JSON(err)
>  }
> ```

I find it better and more modular to use, so we can simplify the overall implementation, for example, reading the fields of the struct, 

```go
func SetDefaultValues(model interface{}) error {
	v := reflect.ValueOf(model)
	if v.Kind() != reflect.Ptr || v.Elem().Kind() != reflect.Struct {
		return errorsUtils.ErrInternalServer
	}
	v = v.Elem()
	for i := 0; i < v.NumField(); i++ {
		field := v.Field(i)
		if field.Kind() == reflect.Ptr && field.IsNil() {
			field.Set(reflect.New(field.Type().Elem()))
		}
	}
	return nil
}
```

create a reader to set "nils" spaces in the fields, to set them with default values, something like this can be used.

**but this is more to work on internal control, structures and requests, to then be able to use this response**

@matdevcoder I will wait for your approval

**Review by matdevcoder — APPROVED** [review]

**Pull request merged**
